processor/src/test/resources/fixtures/org/mapstruct/ap/test/nestedbeans/UserDtoMapperSmartImpl.java

```diff
@@ -0,0 +1,221 @@
+            return null;
+        }
+
+        List<WheelDto> list1 = new ArrayList<WheelDto>();
+        for ( Wheel wheel : list ) {
+            list1.add( wheelToWheelDto( wheel ) );
+        }
+
+        return list1;
+    }
+
+    protected CarDto carToCarDto(Car car) {
+        if ( car == null ) {
+            return null;
+        }
+
+        CarDto carDto = new CarDto();
+
+        carDto.setName( car.getName() );
+        carDto.setYear( car.getYear() );
+        List<WheelDto> list = wheelListToWheelDtoList( car.getWheels() );
+        if ( list != null ) {
+            carDto.setWheels( list );
+        }
+
+        return carDto;
+    }
+
+    protected ExternalRoofType roofTypeToExternalRoofType(RoofType roofType) {
+        if ( roofType == null ) {
+            return null;
+        }
+
+        ExternalRoofType externalRoofType;
+
+        switch ( roofType ) {
+            case OPEN: externalRoofType = ExternalRoofType.OPEN;
+            break;
+            case BOX: externalRoofType = ExternalRoofType.BOX;
+            break;
+            case GAMBREL: externalRoofType = ExternalRoofType.GAMBREL;
+            break;
+            default: throw new IllegalArgumentException( "Unexpected enum constant: " + roofType );
+        }
+
+        return externalRoofType;
+    }
+
+    protected RoofDto roofToRoofDto(Roof roof) {
+        if ( roof == null ) {
+            return null;
+        }
+
+        RoofDto roofDto = new RoofDto();
+
+        roofDto.setColor( String.valueOf( roof.getColor() ) );
+        roofDto.setType( roofTypeToExternalRoofType( roof.getType() ) );
+
+        return roofDto;
+    }
+
+    protected HouseDto houseToHouseDto(House house) {
+        if ( house == null ) {
+            return null;
+        }
+
+        HouseDto houseDto = new HouseDto();
+
+        houseDto.setName( house.getName() );
+        houseDto.setYear( house.getYear() );
+        houseDto.setRoof( roofToRoofDto( house.getRoof() ) );
+
+        return houseDto;
+    }
+
+    protected org.mapstruct.ap.test.nestedbeans.other.WheelDto wheelToWheelDto1(Wheel wheel) {
+        if ( wheel == null ) {
+            return null;
+        }
+
+        org.mapstruct.ap.test.nestedbeans.other.WheelDto wheelDto = new org.mapstruct.ap.test.nestedbeans.other.WheelDto();
+
+        wheelDto.setFront( wheel.isFront() );
+        wheelDto.setRight( wheel.isRight() );
+
+        return wheelDto;
+    }
+
+    protected List<org.mapstruct.ap.test.nestedbeans.other.WheelDto> wheelListToWheelDtoList1(List<Wheel> list) {
+        if ( list == null ) {
+            return null;
+        }
+
+        List<org.mapstruct.ap.test.nestedbeans.other.WheelDto> list1 = new ArrayList<org.mapstruct.ap.test.nestedbeans.other.WheelDto>();
+        for ( Wheel wheel : list ) {
+            list1.add( wheelToWheelDto1( wheel ) );
+        }
+
+        return list1;
+    }
+
+    protected org.mapstruct.ap.test.nestedbeans.other.CarDto carToCarDto1(Car car) {
```

Comment:
Any idea why we don't import `CarDto` here? Probably unrelated to the issue at bay, so could you open an issue for keeping track of it?